src/app/reactUtils.ts

```diff
@@ -12,3 +13,26 @@ export const useDrawFocus = <E extends HTMLElement = HTMLElement>() => {
 
   return ref;
 };
+
+export const onFocusLostHandler = (ref: React.RefObject<HTMLElement>, isEnabled: boolean, cb: () => void) => () => {
+  const el = ref && ref.current;
+  if (!el) { return; }
+
+  const handler = (ev: FocusEvent) => {
+    const relatedTarget = ev.relatedTarget as HTMLElement | null;
+
+    if (!relatedTarget || !elementDescendantOf(relatedTarget, ref.current!)) {
+      cb();
+    }
+  };
+
+  if (isEnabled) {
+    el.addEventListener('focusout', handler as any);
```

Comment:
can we type this handler correctly so we don't need to do `as any`?

Reply:
When I remove `any` then it is throwing:
```
Type '(ev: FocusEvent) => void' is not assignable to type 'EventListener'
```
When I remove import for `import FocusEvent from '@openstax/types/lib.dom';` and use global `FocusEvent` then the handler is okay, but then its throwing an error about:
```
Property 'relatedTarget' does not exist on type 'FocusEvent'
```

Reply:
it is working if I use `FocusEventInit`